### PR TITLE
refactor(crane): use deployment, add RBAC changes

### PR DESCRIPTION
**Pull request by nandor-magyar**
Launching crane as a pod proved to be wrong, reverting to deployment.

### CODE REVIEW

File: golang/pkg/crane/crane.go [hunk 1]
```diff
@@ -48,8 +49,22 @@ func Serve(cfg *config.Configuration) {
...
 
 func grpcClose(ctx context.Context, reason agent.CloseReason) error {
+	cfg := grpc.GetConfigFromContext(ctx).(*config.Configuration)
 	if reason == agent.CloseReason_SHUTDOWN {
 		log.Info().Msg("Remote shutdown requested")
+
+		if cfg.CraneInCluster {
+			log.Debug().Msg("running in cluster, scaling down to 0")
+			err := crux.DeploymentCommand(ctx, &common.ContainerCommandRequest{
+				Container: &common.ContainerIdentifier{
```
Comment (robot9706) [inline golang/pkg/crane/crane.go:59]: Container command is missing?

Comment (nandor-magyar) [reply golang/pkg/crane/crane.go:59]: Oops, fixed.

**Review by robot9706 — APPROVED** [review]

File: golang/pkg/crane/config/config.go [hunk 2]
```diff
@@ -16,6 +16,8 @@ type Configuration struct {
 	KeyIssuer           string        `yaml:"keyIssuer"             env:"KEY_ISSUER"                env-default:"co.dyrector.io/issuer"`
 	KubeConfig          string        `yaml:"kubeConfig"            env:"KUBECONFIG"                env-default:""`
 	TestTimeoutDuration time.Duration `yaml:"testTimeout"           env:"TEST_TIMEOUT"              env-default:"15s"`
+	OwnDeployment       string        `yaml:"ownDeployment"         env:"CRANE_DEPLOYMENT_NAME"`
+	OwnNamespace        string        `yaml:"ownNamescae"           env:"CRANE_DEPLOYMENT_NAMESPACE"`
```
Comment (polaroi8d) [inline golang/pkg/crane/config/config.go:20]: Whhhhhhhhhaaaat is `ownNamescae`?

Comment (nandor-magyar) [reply golang/pkg/crane/config/config.go]: Fixed.

**Pull request merged**